build/Microsoft.DotNet.Cli.Compile.targets

```diff
@@ -0,0 +1,184 @@
+
+  <UsingTask TaskName="FixModeFlags" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="ChangeEntryPointLibraryName" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="RemoveAssetFromDepsPackages" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CleanPublishOutput" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CrossgenDirectory" AssemblyFile="$(CLIBuildDll)" />    
+
+  <Target Name="Compile" DependsOnTargets="Init;
+                                           SetupStage;
+                                           CompileStage;
+                                           BuildProjectsForNuGetPackages;
+                                           GenerateNuGetPackagesArchive">
+  </Target>
+
+  <Target Name="SetupStage">
+    <PropertyGroup>
+      <SrcDirectory>$(RepoRoot)/src</SrcDirectory>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <CompileStageInputs Include="$(SrcDirectory)/**/*.cs" Exclude="$(SrcDirectory)/**/obj/**/*.cs" />
+      <CompileStageInputs Include="$(SrcDirectory)/**/project.json;$(SrcDirectory)/**/project.lock.json" />
+
+      <Stage Include="Stage1">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage1Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage1SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(Stage0Path)/dotnet.exe</DotnetExe>
+      </Stage>
+      <Stage Include="Stage2">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage2Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage2SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(DotnetStage1)</DotnetExe>
+      </Stage>          
+    </ItemGroup>    
+  </Target>    
+
+  <Target Name="CompileStage"
+          Inputs="%(Stage.Inputs)"
+          Outputs="%(StageDirectory)/sdk/$(SdkVersion)/dotnet.dll">
+      <PropertyGroup>      
+        <ExeSuffix Condition=" '$(OSName)' == 'win' ">.exe</ExeSuffix>
+        <ExeSuffix Condition=" '$(OSName)' != 'win' "></ExeSuffix>
+
+        <DynamicLibPrefix Condition=" '$(OSName)' == 'win' "></DynamicLibPrefix>
+        <DynamicLibPrefix Condition=" '$(OSName)' != 'win' ">lib</DynamicLibPrefix>
+
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'win' ">.dll</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'osx' ">.dylib</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(DynamicLibSuffix)' == '' ">.so</DynamicLibSuffix>
+
+        <DotnetHostBaseName>dotnet$(ExeSuffix)</DotnetHostBaseName>
+        <DotnetHostFxrBaseName>$(DynamicLibPrefix)hostfxr$(DynamicLibSuffix)</DotnetHostFxrBaseName>
+        <HostPolicyBaseName>$(DynamicLibPrefix)hostpolicy$(DynamicLibSuffix)</HostPolicyBaseName>
+
+        <StageDirectory>%(Stage.StageDirectory)</StageDirectory>
+        <StageSymbolsDirectory>%(Stage.StageSymbolsDirectory)</StageSymbolsDirectory>
+        <SdkOutputDirectory>$(StageDirectory)/sdk/$(SdkVersion)</SdkOutputDirectory>            
```

Comment:
I am defining as a property here so that I can use it in the Copy task below. In the copy task the Item becomes the files and I can't do %(Stage.StageDirectory). To work around that, I use the property.